torch/package/package_exporter.py

```diff
@@ -231,8 +231,13 @@ def __init__(
         if isinstance(f, (Path, str)):
             f = str(f)
             self.buffer: Optional[BinaryIO] = None
-        else:  # is a byte buffer
+        elif isinstance(f, BinaryIO):  # is a byte buffer
             self.buffer = f
+        else:
+            raise AttributeError(
```

Comment:
```suggestion
            raise TypeError(
```
Confused why this should be an attribute error?

Reply:
@Skylion007 thanks for review, have updated